model_discretization_api.py

```diff
@@ -0,0 +1,209 @@
+'''
+A framework for efficient model image rasterization in astropy. 
+
+This is a GSoC 2015 project and is therefore on a relatively short time frame. 
+We focus first on developing a robust solution to a simple but common use case that 
+is extensible to future implementation of more complex models and solutions. 
+
+Use case: Generate a sky model image of discrete sources, e.g. something like 
+photutils.datasets.make_100gaussians_image(), but faster and convolving a source model
+with an instrument PSF. 
+
+The primary challenge- and motivation- is maximizing speed while maintainging precision. A 
+straightforward approach is to use oversampling within a bounded area around the 
+source containing the majority of its flux (e.g. 5-sigma) for discretization and integration. 
+
+The key steps are:
+1. Model generation
+	- Convolution of a source model with an instrument PSF model
+	- Fitting the convolved model to image data if required
+2. Model discretization/Image rasterization
+	- Integration of the model over pixel areas
+	- Reprojection if required
+
+The pieces necessary are largely in existence. Relevant packages 
+include astropy.modeling, astropy.convolution, astropy.nddata, sherpa, galfit, reproject,
+... others.
+
+Outlined here is a rough draft of the proposed API for community discussion. 
+
+'''
+
+######################
+# ConvolvedModel API #
+######################
+
+from astropy.modeling import ConvolvedModel
+from astropy.modeling.models import Box2D, Lorentz1D, Gaussian2D
+from astropy.convolution import Gaussian2DKernel, Gaussian1DKernel
+
+# Initiate a ConvolvedModel instance, takes source model and convolution kernel
+# like:
+source = Box2D(1, 0, 0, 1, 1)
+psf = Gaussian2DKernel(1)
+convolved_model = ConvolvedModel(source, psf, mode='oversample', factor=10)
+
+# mode will be used both for convolution and later integration
+# if a different mode is desired for convolution it should be passed in the psf
+
+# support of 1D models
+source = Lorentz1D(1, 0, 0, 1, 1)
+psf = Gaussian1DKernel(1)
+convolved_model = ConvolvedModel(source, psf)
+
+# should there be a Convolved1DModel and Convolved2DModel?
+# should this be an extension of the CompoundModel class?
```

Comment:
To me what a convolution operator implies in this context is that the form of the input model must be returned as a regularly sampled set of values (effectively table look up) so that a discrete convolution can be applied. For example, if the input model is purely analytic (or even table look-up on a different grid, or irregular grid), the model must be evaluated on a grid that the convolved can use. In effect the convolver has to turn the input into a lookup table model and operate on that. The convolver needs info on the sampling interval and range that it is expected to deal with, and uses that to evaluate the input to that. Is there any other way to deal with it?